### ISSUE TITLE
Is writing to file operation thread-safe?

### ISSUE THREAD

**Issue by ScottShao** [buckyroberts/Spider#26]
Is it possible several threads are writing to the file using update_files at the same time?

**Comment by buckyroberts** [comment]
Yep, a fix is needed to either use locking or to use separate files per thread then merge at the end. Feel free to submit a pull request if you can think of a solution that works.

**Comment by ghost** [comment]
The project has been moved to https://github.com/AbdulSheikh/Spider . This project is no longer supported In this github account.
